posts/2021-08-30-binding-app-to-postgresql-db-with-odo-and-ol-stack.adoc

```diff
@@ -0,0 +1,212 @@
+
+
+Install the needed operators.
+
+If you have the right priviledges, you can use the OpenShift console to install these operators. You do this by navigating to `Operators->OperatorHub` and searching/selecting/installing a couple of operators from the catalog.
+
+- Service Binding Operator (provided by Red Hat)
+
+Use the OpenShift console (`Operators->OperatorHub`) to install this operator. Be sure to install version 0.9.1 and above.
+
+- PostgreSQL Operator (provided by Dev4Ddevs.com)
+
+Before installing this operator, create the project/namespace under which the application will be deployed. This operator is namespace scoped, so it needs to be installed in the same namespace where the application will be deployed.
+
+[source,sh]
+----
+$ odo project create service-binding-demo
+----
+
+Use the OpenShift console (`Operators->OperatorHub`) to install this operator. Be sure to pick `service-binding-demo` as the namespace in which to install it.
+
+Clone the application repository.
+
+[source,sh]
+----
+$ git clone https://github.com/OpenLiberty/application-stack-samples.git && \
+  cd application-stack-samples/jpa
+----
+
+Create a Java Open Liberty component.
+
+If you want the application to be built and deployed using Maven:
+
+[source,sh]
+----
+$ odo create java-openliberty mysboproj
+----
+
+If you want the application to be built and deployed using Gradle:
+
+[source,sh]
+----
+$ odo create java-openliberty-gradle mysboproj
+----
+
+Push the application to the cluster.
+
+[source,sh]
+----
+$ odo push
+----
+
+So far, the application has been deployed on your cluster. However, the application is not ready to be used yet because the application does not have the data needed to access the database, and the PostgreSQL database instance has not been created. Let's create the database instance next.
+
+Display the service providers and services available in the cluster.
+
+[source,sh]
+----
+odo catalog list services
+Services available through Operators
+NAME                                CRDs
+postgresql-operator.v0.1.1          Backup, Database
+service-binding-operator.v0.9.1     ServiceBinding, ServiceBinding
+----
+
+Generate the yaml config of the Database resource/service provided by the postgresql-operator.v0.1.1 operator and store it in a file. We do this because we need to edit that file to modify some access information and to tell the Service Binding Operator, in annotation form, what data provided by this service needs to be bound to the application.
+
+[source,sh]
+----
+odo service create postgresql-operator.v0.1.1/Database --dry-run > db.yaml
+----
+
+Modify the database name, user, and password values under the spec section in db.yaml
+
+[source,sh]
+----
+  databaseName: "sampledb"
+  databasePassword: "samplepwd"
+  databaseUser: "sampleuser"
+----
+
+Add the needed annotations under the metadata section in db.yaml. 
+
+[source,sh]
+----
+metadata:
+  name: sampledatabase
```

Comment:
You don't explicitly say it but you're implicitly saying to change the metadata.name...since the default is 'database'.   This name is used as the target of the link command so if they don't that will fail later.